pkg/system/phase4_configuring.go

```diff
@@ -1323,6 +1327,82 @@ func (r *Reconciler) ReconcileOBCStorageClass() error {
 	return nil
 }
 
+// ReconcileBucketLoggingPVC ensures the bucket logging PVC is properly configured for 'guaranteed' logging
+func (r *Reconciler) ReconcileBucketLoggingPVC() error {
+	log := r.Logger.WithField("func", "ReconcileBucketLoggingPVC")
+
+	// Return if Bucket logging type is either not specified or not set to 'guaranteed'
+	if r.NooBaa.Spec.BucketLogging.LoggingType != nbv1.BucketLoggingTypeGuaranteed {
+		log.Info("Bucket logging type is either not specified or not set to 'Guaranteed'. skipping ReconcileBucketloggingPVC")
+		return nil
+	}
+
+	util.KubeCheck(r.BucketLoggingPVC)
+	if err := r.prepareBucketLoggingPVC(); err != nil {
+		return err
+	}
+	r.Own(r.BucketLoggingPVC)
+
+	if r.BucketLoggingPVC.UID != "" {
+		if r.BucketLoggingPVC.Status.Phase == corev1.ClaimPending {
+			log.Infof("BucketLoggingPVC %s is in pending state. Investigating the reason and updating pvc", r.BucketLoggingPVC.Name)
+			// TODO: fetch reason and return only for the specific error, like RWX mode not supported by provisioner
+
+			// events := &corev1.EventList{}
+			// listOpts := &client.ListOptions{
+			// 	Namespace:     r.BucketLoggingPVC.Namespace,
+			// 	FieldSelector: fields.OneTermEqualSelector("involvedObject.uid", string(r.BucketLoggingPVC.UID)),
+			// }
+			// if err := r.Client.List(r.Ctx, events, listOpts); err != nil {
+			// 	return err
+			// }
+
+			log.Infof("Deleting and recreating the pending BucketLoggingPVC %s", r.BucketLoggingPVC.Name)
+			if err := r.Client.Delete(r.Ctx, r.BucketLoggingPVC); err != nil {
+				return err
+			}
+			if err := r.Client.Create(r.Ctx, r.BucketLoggingPVC); err != nil {
+				return err
+			}
```

Comment:
Could you explain this? 
Can't a PVC be pending and have data on it?

Reply:
Thanks, I will take care of it.

Actually the idea here is to update the PVC if new StorageClass was provided by user and PVC is in pending state from the start(PV not provisioned). Although we will return before it on the basis of specific error if it is possible.